src/Sylius/Bundle/PayumBundle/Validator/GatewayFactoryExistsValidator.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\PayumBundle\Validator;
+
+use Sylius\Bundle\PayumBundle\Model\GatewayConfigInterface;
+use Sylius\Bundle\PayumBundle\Validator\Constraints\GatewayFactoryExists;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+final class GatewayFactoryExistsValidator extends ConstraintValidator
+{
+    public const OFFLINE_GATEWAY_FACTORY = 'offline';
+
+    /** @param array<string, string> $factoryNames */
+    public function __construct(private array $factoryNames)
+    {
+    }
+
+    /** @param string|null $value */
+    public function validate($value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof GatewayFactoryExists) {
+            throw new UnexpectedTypeException($constraint, GatewayFactoryExists::class);
+        }
+
+        if ($value === self::OFFLINE_GATEWAY_FACTORY || $value === null || $value === '') {
+            return;
+        }
```

Comment:
Maybe we should replace the `$value === self::OFFLINE_GATEWAY_FACTORY` with
```suggestion
        if (in_array($value, $this->alwaysExistingGateways) || $value === null || $value === '') {
            return;
        }
```

Where `alwaysExistingGateways` is passed via the constructor. I can't come up with any example, but I feel someone might have some case behaving similar to the `Offline` one. @NoResponseMate wdyt?

Reply:
I'm doubtful it'd ever be used, but it's a small change and in the off-chance, there'd be an easy extension point already in place, so why not.

Reply:
I would be after maybe $this->gatewaysWithoutConfiguration or something like that, but on the other hand why would they use different gateway if it does not differ from the 'offline' one?

Reply:
Nevermind, it is registered along with other factories, this check was valid in the previous response, now we can get rid of it 😅